jupyter_server/services/contents/handlers.py

```diff
@@ -238,12 +238,14 @@ async def _new_untitled(self, path, type="", ext=""):
         validate_model(model)
         self._finish_model(model)
 
-    async def _save(self, model, path):
+    async def _save(self, model, path, require_hash):
         """Save an existing file."""
         chunk = model.get("chunk", None)
         if not chunk or chunk == -1:  # Avoid tedious log information
             self.log.info("Saving file at %s", path)
-        model = await ensure_async(self.contents_manager.save(model, path))
+        model = await ensure_async(
+            self.contents_manager.save(model, path, require_hash=require_hash)
```

Comment:
I believe this should fix the backward compatibility @Zsailer had in mind. Import needs to be moved, to top but GitHub UI does not allow to add a suggestion on non-changed part of the file.

```suggestion
            import inspect
            save = self.contents_manager.save
            signature = inspect.signature(save)
            parameters = signature.parameters
            has_kwargs = any(p.kind == p.VAR_KEYWORD for p in parameters)
            if "require_hash" in parameters or has_kwargs:
                return save(model, path, require_hash=require_hash)
            else:
                return save(model, path)
```